datafusion/expr/src/type_coercion/functions.rs

```diff
@@ -450,19 +473,7 @@ fn coerced_from<'a>(
         {
             Some(type_into.clone())
         }
-        // More coerce rules.
-        // Note that not all rules in `comparison_coercion` can be reused here.
-        // For example, all numeric types can be coerced into Utf8 for comparison,
-        // but not for function arguments.
-        _ => comparison_binary_numeric_coercion(type_into, type_from).and_then(
```

Comment:
This logic is introduced in #9459, so I think it is safe to remove together with this PR

Reply:
cc @viirya